pkg/cli/admin/ocpcertificates/regeneratemco/command.go

```diff
@@ -0,0 +1,297 @@
+			Client:        clientset.CoreV1(),
+			EventRecorder: recorder,
+		},
+		nil, // no operatorclient needed
+		recorder,
+	)
+
+	ch := make(chan struct{})
+	inf.Start(ch)
+	inf.WaitForCacheSync(ch)
+
+	ctx := context.WithValue(context.Background(), certrotation.RunOnceContextKey, true)
+	syncCtx := factory.NewSyncContext(mcsName, recorder)
+	if err := cont.Sync(ctx, syncCtx); err != nil {
+		return err
+	}
+
+	fmt.Println("Successfully rotated MCS CA + certs.")
+
+	// Redeploy MCS. This will eventually not be needed, see: https://github.com/openshift/machine-config-operator/pull/3744
+	mcoPods := clientset.CoreV1().Pods(mcoNamespace)
+	mcsPods, err := mcoPods.List(ctx, metav1.ListOptions{
+		LabelSelector: mcsLabelSelector,
+	})
+	if err != nil {
+		return fmt.Errorf("cannot get MCS pods: %w", err)
+	}
+	for _, pod := range mcsPods.Items {
+		err := mcoPods.Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("cannot delete MCS pod %s: %w", pod.Name, err)
+		}
+	}
+
+	// TODO maybe add a watcher to make sure the MCS daemonset is ready here
+
+	// Get the updated CA cert
+	mcoSecrets := clientset.CoreV1().Secrets(mcoNamespace)
+	mcsSecret, err := mcoSecrets.Get(ctx, newMCSCASecret, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot read MCS secret: %w", err)
+	}
+
+	caCert, exists := mcsSecret.Data[CASecretCertName]
+	if !exists {
+		return fmt.Errorf("could not find MCS CA cert at %s after rotation", CASecretCertName)
+	}
+	encodedCACert := base64.StdEncoding.EncodeToString(caCert)
+
+	if o.outputNewCert {
+		fmt.Printf("New Certificate:\n%s\n", encodedCACert)
+	}
+
+	if !o.modifyUserData {
+		return nil
+	}
+
+	// Get user-data-secret
+	mapiSecrets := clientset.CoreV1().Secrets(mapiNamespace)
+	secretList, err := mapiSecrets.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot list MAO secrets: %w", err)
+	}
+	for _, secret := range secretList.Items {
+		// These two are managed by the MCO but unused. Skip them since the MCO will write them back.
+		if secret.Name == mcoManagedWorkerSecret || secret.Name == mcoManagedMasterSecret {
+			continue
+		}
+
+		// These secrets don't really have a label or not, so the determining factor is if they:
+		// 1. have a userData field
+		// 2. is an ignition config
+		userData, exists := secret.Data[userDataKey]
+		if !exists {
+			continue
+		}
+		// userData is an ignition config. To save the effort of multiple-version parsing, just parse it as a json
+		var userDataIgn interface{}
+		err = json.Unmarshal(userData, &userDataIgn)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal decoded user-data to json (secret %s): %w", secret.Name, err)
+		}
+		ignMap := userDataIgn.(map[string]interface{})
+		ign, isIgn := ignMap["ignition"].(map[string]interface{})
+		if !isIgn {
+			// Didn't find "ignition in user-data, warn but continue"
+			fmt.Printf("Unable to find ignition in user-data, skipping secret %s\n", secret.Name)
```

Comment:
Fprintf to the iostream.  Doing that allows composition and unit testing for output if desired at some point.